tests/test_loops.py

```diff
@@ -70,6 +71,47 @@ def loop_args():
     return lit_api_mock, requests_queue
 
 
+class TestQueue(Queue):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._sentinel_seen = False
+
+    def get(self, timeout=None):
+        if self._sentinel_seen:
+            raise Empty  # Simulate queue being empty after sentinel
+        item = super().get(timeout=timeout)
+        # Sentinel: (None, None, None, None)
+        if item == (None, None, None, None):
+            self._sentinel_seen = True
+            raise KeyboardInterrupt  # Triggers loop exit in your code
+        return item
```

Comment:
love this! in future, maybe we can add something directly in the Loops for test mode so that we don't have to do this everywhere.

Reply:
sure.